tests/testsuite/warn_on_failure.rs

```diff
@@ -90,17 +88,14 @@ fn no_error_or_warning_on_bin_failure() {
         .cargo("build")
         .with_status(101)
         .with_stdout_does_not_contain("hidden stdout")
-        .with_stderr_data(str![[r#"
-[UPDATING] `dummy-registry` index
-[LOCKING] 2 packages to latest compatible versions
-[DOWNLOADING] crates ...
-[DOWNLOADED] bar v0.0.1 (registry `dummy-registry`)
-[COMPILING] bar v0.0.1
-[ERROR] invalid output in build script of `bar v0.0.1`: `cargo::error=Hello! I'm an error. :)`
-Unknown key: `error`.
-See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script for more information about build script outputs.
-
-"#]])
+        .with_stderr_does_not_contain("hidden stderr")
+        .with_stderr_does_not_contain(&format!("[ERROR] {}", ERROR))
+        .with_stderr_does_not_contain(&format!("[WARNING] {}", WARNING1))
+        .with_stderr_does_not_contain(&format!("[WARNING] {}", WARNING2))
+        .with_stderr_contains("[UPDATING] `[..]` index")
+        .with_stderr_contains("[DOWNLOADED] bar v0.0.1 ([..])")
+        .with_stderr_contains("[COMPILING] bar v0.0.1")
+        .with_stderr_contains("[COMPILING] foo v0.0.1 ([..])")
```

Comment:
Why was the test style changed?  We are trying to move away from `does_not_contain` and `contains`

Reply:
It's actually not changed overall. I upgraded the style in the first commit, but left it as it was in this one, so that the PR as a whole only changes one line. But I can update the styles overall.

Reply:
Ah, it was switched back and forth